### ISSUE TITLE
Favorites UX improvements

### ISSUE THREAD

**Issue by Kuroonehalf** [Bionus/imgbrd-grabber#1465]
When I first installed Grabber I read its documentation and tested out the favorites feature, but then promptly forgot about it, and as of lately have spent an inordinate amount of time trying to figure out just how the in the devil you can favorite something without resorting to the documentation. This points to a glaring design problem.

So currently, according to the documentation, the only way to favorite a multi-tag search is to right click on the search field and access the favorites options in that context menu. This is incredibly unintuitive and obtuse because right-clicking text input fields is pretty much always reserved for text editing context options. For example when I right click the input field in a google search I get options about copying/pasting/cutting text, undoing, and spell-checking. This is the same in pretty much every other search input field, and is how users will naturally expect Grabber's to behave as well.
Instead, there's other places where I would expect this option to be present, and where it is not. Here are them in decreasing order of importance:
- Right clicking a tab. It's how the interface distinguishes searches, and it makes the most sense for the option to be accessible there.
- In the top menu. For people who might not be as savvy to think that the above is an option, having a Favorites item in the top menu would be the most logical place to expect to find this. From there, have an option to add the current tab's query to the favorites.
- Through the favorites window. Have a button at the bottom of the window and/or an option in the context menu. Additionally, you should be able to edit a favorited search's options from here. It feels unnecessary to have to move to the favorites tab to do this.
- Through the favorites tab. In the same way that you can edit or delete a favorited search in the favorites tab, it only makes sense to be able to add them from there as well.

Ideally I'd like to see all of these implemented, as they all contribute to the user experience in using this tool.

**Comment by Bionus** [comment]
> So currently, according to the documentation, the only way to favorite a multi-tag search is to right click on the search field and access the favorites options in that context menu. This is incredibly unintuitive and obtuse

Yes, as you said when I made the favorites system a few years ago, it was designed for single tags (favorite options for tags are much more intuitive), which is why multi-tag favorites are kind of second-class citizen in the current Grabber. 😞 

> because right-clicking text input fields is pretty much always reserved for text editing context options.
> For example when I right click the input field in a google search I get options about copying/pasting/cutting text, undoing, and spell-checking. This is the same in pretty much every other search input field, and is how users will naturally expect Grabber's to behave as well.

I personally disagree. While that is true for basic text fields, once an input gets syntax highlighting as well as auto-complete, I don't find it chocking to have its context menu overhauled as well.

> Instead, there's other places where I would expect this option to be present, and where it is not.

This I agree. The fact that it is available *only* in the context menu is an issue. Browsers usually have an icon as well as an action in a "Bookmarks" menu.

> Right clicking a tab. It's how the interface distinguishes searches, and it makes the most sense for the option to be accessible there.

Are you talking about the whole tab (in blank spaces), or only the part in the tab bar at the top? (where the tab name is located)

> In the top menu. For people who might not be as savvy to think that the above is an option, having a Favorites item in the top menu would be the most logical place to expect to find this. From there, have an option to add the current tab's query to the favorites.

👍 

Now to think where exactly given it does not fit the "File" menu, and a "Favorites" menu with nothing but one action inside sounds overkill. 🤔 

> Through the favorites window. Have a button at the bottom of the window and/or an option in the context menu. Additionally, you should be able to edit a favorited search's options from here. It feels unnecessary to have to move to the favorites tab to do this.

Are you talking about the small dockable flying box that contains the list of favorites?

![image](https://user-images.githubusercontent.com/882719/46910916-3649b900-cf4e-11e8-99c7-40cc977660e7.png)

I feel that indeed the right click on favorites could show an "Edit" as well as a "Remove" action (instead of the useless "Copy Link Location" default)

> Through the favorites tab. In the same way that you can edit or delete a favorited search in the favorites tab, it only makes sense to be able to add them from there as well.

Some kind of button that when clicked prompt the favorite edit window? (except it would be empty and titled "add a favorite")

**Comment by Penguin-Guru** [comment]
All of these seem like great suggestions to me.

I kind of like the idea of using context menus for things other than just text editing options. It seems like a waste that this is so unused in other programs. If context menus get too cluttered, just organise them. I see some context menus already have nice, expandable sub-menus that work well. You could either have one sub-menu for text options or just sort all the text options at the top and use a horizontal bar to divide them from everything else.

About putting a favourites menu in the top menu bar, I think it's a good idea even if there would only be one button in it for now. It seems likely that there will be other things that would be useful there, and it could always be removed later or controlled by a setting. In my opinion, favourites are one of the main reasons people use a program like this so it makes sense for that to be on the main/application menu bar. As for placement, I would personally put it on the right end so all the standard menus could appear where people expect them. Separating it with a vertical bar might be good too, I can't say without seeing it.

Slight tangent: I think it might be worth moving the Help menu to the far right side of the bar since its contents aren't really related to any application settings or functions. I think this is a pretty common placement for help buttons but they usually seem to use an icon, like a question mark, so it doesn't look like the main menu spans the whole window width.
